networkit/cpp/distance/HopPlotApproximation.cpp

```diff
@@ -66,7 +66,7 @@ void HopPlotApproximation::run() {
         }
     });
     // at zero distance, all nodes can only reach themselves
-    hopPlot[0] = 1/G->numberOfNodes();
+    hopPlot[0] = 1.0 / static_cast<double>(G->numberOfNodes());
```

Comment:
This looks somewhat strange. Is this what the algorithm intended to do?

Reply:
I don't know exactly but that line assigns `hopPlot[0]` to zero, which is probably not wanted.

Reply:
The change should be right. In the code the hop plot at zero (given as fractional number) is `n` (since all nodes can only reach themselves) divided by all possible combinations of pairs `n^2`.